src/store.js

```diff
@@ -182,6 +186,12 @@ function EvolvStore(options) {
       return;
     }
 
+    if (LOCKED) {
+      return;
+    }
+
+    LOCKED = true;
```

Comment:
What requires the need for a mutex like this? I can't find any asynchronous statements within this functions.

Reply:
The updating of the context from within reevaluateContext causes reevaluateContext to be called

Reply:
This is due to the async listeners on the context object itself. I think in order to do this without the use of mutexes we would have reevaluate the context synchronously within the methods that update the context. Which could be done, but may result in a significant change.